packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -238,7 +239,9 @@ class AndroidDevice extends Device {
     if (!_checkForSupportedAdbVersion() || !_checkForSupportedAndroidVersion())
       return false;
 
+    Status status = logger.startProgress('Installing ${apk.apkPath}...');
     String installOut = runCheckedSync(adbCommandForDevice(<String>['install', '-r', apk.apkPath]));
```

Comment:
This `runCheckedSync` may mean that we don't get a spinner during the install. We'd need to change this to a async process call (and change the containing method signature to a Future result).